admin/src/components/mailing/ComposeEmailModal.tsx

```
@@ -0,0 +1,112 @@
+                onClick={() => setShowPreview(!showPreview)}
+                className="text-xs text-blue-600 hover:text-blue-800 flex items-center gap-1"
+              >
+                <Eye className="w-3 h-3" />
+                {showPreview ? 'Edit' : 'Preview'}
+              </button>
+            </div>
+
+            {showPreview ? (
+              <div
+                className="border border-gray-300 rounded-md p-3 min-h-[200px] text-sm bg-white prose prose-sm max-w-none"
+                dangerouslySetInnerHTML={{ __html: bodyHtml }}
```

Biome (2.3.14)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)